File: algorithms/math/recursion/fibonacci-sequence.js
// Problem: Give a number 'n', find the nth elements of the Fibonacci sequence

function recursiveFibonacci(n) {
  // base case
  if (n < 2) {
    return n;
  }
  return recursiveFibonacci(n - 1) + recursiveFibonacci(n - 2);
}

console.log(recursiveFibonacci(0)); //   = 0
console.log(recursiveFibonacci(1)); //   = 1
console.log(recursiveFibonacci(6)); //   = 8

// Big-O = O(2^n) --- terrible!!

File: algorithms/math/python/recursion/fibonacci-sequence.py
# O(n^2)
def recursive_fibonacci(n):
    # base case
    if n < 2:
        return n
    return recursive_fibonacci(n - 1) + recursive_fibonacci(n - 2)


print(recursive_fibonacci(0))  # = 0
print(recursive_fibonacci(1))  # = 1
print(recursive_fibonacci(6))  # = 8
